pipeline/frontend/yaml/compiler/compiler.go

```diff
@@ -85,6 +85,12 @@ func New(opts ...Option) *Compiler {
 func (c *Compiler) Compile(conf *yaml.Config) *backend.Config {
 	config := new(backend.Config)
 
+	if !conf.When.Match(c.metadata, true) {
+		// This pipeline does not match the configured filter so return an empty config and stop further compilation.
+		// An empty pipeline will just be skipped and wont be shown in the UI as well.
```

Comment:
```suggestion
		// An empty pipeline will just be skipped completely.
```

It wont be saved to db, shown in UI, ... => that's what skips means ;)